queue_job/models/queue_job.py

```diff
@@ -235,6 +238,70 @@ def requeue(self):
         self._change_job_state(PENDING)
         return True
 
+    def button_terminate(self):
+        self.terminate()
+        return {
+            "type": "ir.actions.client",
+            "tag": "reload",
+        }
+
+    def terminate(self, raise_error=True):
+        """Attempt to kill the backend process related to the running job."""
+        error_jobs = self.env["queue.job"]
+        for record in self:
+            # Check if the job can be terminated
+            if record.state != STARTED:
+                error_message = _(
+                    f"Unable to terminate job {record.uuid} because it is not started"
+                )
+                _logger.warning(error_message)
+                if raise_error:
+                    raise exceptions.UserError(error_message)
+                continue
+
+            elif not record.db_txid:
+                error_message = _(
+                    f"Unable to terminate job {record.uuid}. Missing 'db_txid'",
+                )
+                _logger.warning(error_message)
+                if raise_error:
+                    raise exceptions.UserError(error_message)
+                continue
+
+            # Compare xact_start and job date_started as reference
+
+            self.env.cr.execute(
+                """
+                SELECT pid FROM pg_stat_activity
+                WHERE backend_xid = %(db_xid)s
+                AND state != 'idle';
+                """,
+                {"db_xid": record.db_txid}
+            )
+
+            res = self.env.cr.fetchone()
+            if res:
+                try:
+                    _logger.warning(
+                        "Terminating backend process with transaction %s for job %s",
+                        record.db_txid,
+                        record.uuid,
+                    )
+                    self.env.cr.execute(
+                        "SELECT pg_terminate_backend(%(backend_pid)s);",
```

Comment:
At this point, backend_pid can already be running another transaction. You have to `select pg_terminate_backend(pid)` inline in your query on pg_stat_activity.